libfreefare/freefare_internal.h

```diff
@@ -215,6 +210,13 @@ struct mifare_desfire_tag {
     uint32_t selected_application;
 };
 
+struct mifare_key_deriver {
```

Comment:
The `struct mifare_key_deriver` seems to be accessed only from the `mifare_key_deriver.c` file.  Since the struct is not accessed to from the tests, maybe you shloud consider moving the struct definition to the `mifare_key_deriver.c` file?

Reply:
No problem.

Reply:
The contents of this struct are needed by the test case code.